include/xgboost/context.h

```diff
@@ -262,7 +269,6 @@ struct Context : public XGBoostParameter<Context> {
    */
   template <typename CPUFn, typename CUDAFn, typename SYCLFn>
   decltype(auto) DispatchDevice(CPUFn&& cpu_fn, CUDAFn&& cuda_fn, SYCLFn&& sycl_fn) const {
-    static_assert(std::is_same_v<std::invoke_result_t<CPUFn>, std::invoke_result_t<CUDAFn>>);
```

Comment:
why?

Reply:
In case of non-sycl device, the `DispatchDevice` with this assert will be called. So, it was just a duplicated assert.